pilot/pkg/networking/core/route/route.go

```diff
@@ -1626,16 +1628,18 @@ func cutPrefix(s, prefix string) (after string, found bool) {
 	return s[len(prefix):], true
 }
 
-func CheckAndGetInferencePoolConfig(virtualService config.Config) string {
+func CheckAndGetInferencePoolConfigs(virtualService config.Config) map[string]string {
 	vs := virtualService.Spec.(*networking.VirtualService)
-	var infPoolConfig string
+	infPoolConfigs := map[string]string{}
 	for _, httpRoute := range vs.Http {
 		routeNameParts := strings.Split(httpRoute.Name, "%%")
 		if len(routeNameParts) > 1 {
```

Comment:
nit: we are checking its at least 2 items, then index the 3rd which may not be safe (unchanged in this PR, just noticed)

Reply:
thanks - https://github.com/istio/istio/pull/55705